File: Flatland-Space-Stations.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    // Complete the flatlandSpaceStations function below.
    static int flatlandSpaceStations(int n, int[] arr) {
        Array.Sort(arr);
        int maxDistance = arr[0];

        for (int i = 1; i < arr.Length; i++)
        {
            int distance = (arr[i] - arr[i - 1]) / 2;
            if (maxDistance < distance)
            {
                maxDistance = distance;
            }
        }

        int lastGap = (n - 1) - arr[arr.Length - 1];

        return (maxDistance < lastGap) ? lastGap : maxDistance;

    }

    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string[] nm = Console.ReadLine().Split(' ');

        int n = Convert.ToInt32(nm[0]);

        int m = Convert.ToInt32(nm[1]);

        int[] c = Array.ConvertAll(Console.ReadLine().Split(' '), cTemp => Convert.ToInt32(cTemp));
        int result = flatlandSpaceStations(n, c);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Flatland-Space-Stations.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the flatlandSpaceStations function below.
    static int flatlandSpaceStations(int n, int[] arr) {
        Arrays.sort(arr);
        int maxDistance = arr[0];

        for (int i = 1; i < arr.length; i++) {
            int distance = (arr[i] - arr[i - 1]) / 2;
            if (maxDistance < distance) {
                maxDistance = distance;
            }
        }

        int lastGap = (n - 1) - arr[arr.length - 1];

        return (maxDistance < lastGap) ? lastGap : maxDistance;

    }
}
